pynopegl/init_header.py

```diff
@@ -195,8 +195,6 @@ def _args_setter(self, cython_setter, param_name, *args):
         return cython_setter(param_name, args)
 
     def _add_nodes(self, param_name, *nodes):
-        if hasattr(nodes[0], "__iter__"):
-            raise Exception(f"add_{param_name}() takes elements as positional arguments, not list")
```

Comment:
This is a bit tricky for several reasons:
- This check is here to prevent code like `group.add_children([a, b, c])` which is a common mistake
- removing this check indeed allows empty sized `nodes` array, but then in `_pynopegl.pyx` you'll get a `calloc(0, ...)`, which is implementation defined (could either return a pointer or a `NULL`), which ends up making `ngl.Group(children=[])` non deterministic (depending on the platform)
- The f64s handling should be consistent with that logic

So instead, may I suggest instead adding something along the lines of `if not nodes: return 0` (and same with `f64s` below)?

Reply:
Right, changed it.